wolfcrypt/src/ecc.c

```diff
@@ -6298,8 +6279,11 @@ int wc_ecc_import_x963_ex(const byte* in, word32 inLen, ecc_key* key,
     in += 1;
 
 #ifdef WOLFSSL_ATECC508A
-    /* populate key->pubkey_raw */
-    XMEMCPY(key->pubkey_raw, (byte*)in, sizeof(key->pubkey_raw));
+    /* For SECP256R1 only save raw public key for hardware */
+    if (curve_id == ECC_SECP256R1 && !compressed &&
+                                            inLen <= sizeof(key->pubkey_raw)) {
+        XMEMCPY(key->pubkey_raw, (byte*)in, sizeof(key->pubkey_raw));
```

Comment:
If inLen is less than sizeof(key->pub key_raw) reading the full sizeof bytes is an error.  Also, what about the case of inLen greater than or a different curve;  shouldn't we detect that error and tell the user?